dask/array/core.py

```diff
@@ -2231,10 +2231,15 @@ def astype(self, dtype, **kwargs):
             By default, astype always returns a newly allocated array. If this
             is set to False and the `dtype` requirement is satisfied, the input
             array is returned instead of a copy.
+        order : {‘C’, ‘F’, ‘A’, ‘K’}, optional
+            Memory layout. ‘A’ and ‘K’ depend on the order of input array a.
+            ‘C’ row-major (C-style), ‘F’ column-major (Fortran-style) memory
+            representation. ‘A’ (any) means ‘F’ if a is Fortran contiguous, ‘C’
+            otherwise ‘K’ (keep) preserve input order. Defaults to ‘C’.
```

Comment:
Great idea, we should absolutely add a note. This is not a good place for it, however. It's not specific to `astype`, you can see from the bug example in https://github.com/dask/dask/issues/9316 that it can happen only using `from_array`.

So maybe we should put the note in the docstring for `from_array`? To be honest, I'm not really sure if that is the best place, because it might affect dask arrays more broadly than that. But I don't have a better idea right now, and I think more people might see the note there.

Reply:
Oh hang on, I see:
1. Julia has already added that note to `from_array`
2. Maybe it's better to take the order kwarg out of the docstring here entirely, because it doesn't actually do what I say it does. That would have the benefit of other people's code not crashing, but also wouldn't encourage people to do this thing.

```python
import dask.array as da
import numpy as np

numpy_array = np.array([[1,2,3,4],[5,6,7,8],[9,10,11,12]], order="C")
print(numpy_array.flags)
#  C_CONTIGUOUS : True
#  F_CONTIGUOUS : False
#  OWNDATA : True
#  WRITEABLE : True
#  ALIGNED : True
#  WRITEBACKIFCOPY : False
#  UPDATEIFCOPY : False

dask_array = da.from_array(numpy_array, chunks=(2,2)) # we already know from_array doesn't preserve order
# astype should change the contiguous order now with this PR, but it doesn't
dask_array = dask_array.astype(numpy_array.dtype, order="F")
dask_array.compute().flags
#  C_CONTIGUOUS : True
#  F_CONTIGUOUS : False
#  OWNDATA : True
#  WRITEABLE : True
#  ALIGNED : True
#  WRITEBACKIFCOPY : False
#  UPDATEIFCOPY : False
```

Reply:
Ok, I've decided to simply ignore the `order` kwarg, and add a note to the docstring that says that's what we are doing.